R/SpatialExperiment-methods.R

```diff
@@ -175,8 +175,10 @@ setReplaceMethod("spatialData",
 setReplaceMethod("spatialData", 
     c("SpatialExperiment", "NULL"),
     function(x, value) {
+        # @<- is required here to enforce replacement because
+        # spatialDataNames are protected during colData replacement
         i <- spatialDataNames(x)
-        colData(x)[i] <- NULL
```

Comment:
Just a question, so we decided to entirely drop the `colData` and the `spatialData`?
Sorry, I don't remember the entire discussion ...

Reply:
Hm, I think we discussed to separate the two. So one can drop the `spatialData` (which are just other cell metadata (`colData`), e.g. `in_tissue` might be irrelevant at some point). However, `spatialCoords` are kept separate from this, similar to a `reducedDim`. I think that makes sense - but yes, it's a different way of thinking than what we had previously. However, given that most computations will eventually run on `spatialCoords` (plus other non-`spatialData` cell metadata), I think this is sensible.

Reply:
ok for the spatialCoords, it makes sense. 
But the instruction `x@colData[i] <- NULL` drops also the `spatialData` right?

Reply:
Yes, for the `spatialData()<-NULL` replacement methodm anything specified as `spatialDataNames` in `int_metadata()$spatialDataNames` will be dropped. Does that make sense or did you mean something else?